src/Dscp/Resource/Functions.hs

```diff
@@ -0,0 +1,67 @@
+
+module Dscp.Resource.Functions
+       ( handleComponentErrors
+       , runComponentR
+       , runResourceAllocation
+       ) where
+
+import Control.Monad.Component (ComponentError (..), ComponentM, runComponentM)
+import qualified Data.Text.Prettyprint.Doc as Doc (pretty)
+import Fmt ((+||), (||+))
+import Loot.Log (logError)
+
+import Dscp.Launcher.Rio (runRIO)
+import Dscp.Resource.Class (InitContext (..), InitParams (..))
+import Dscp.Resource.Logging (allocLogging)
+
+-- | Catches and prints possible synchronous exceptions thrown
+-- in 'runComponentM'. This includes both resource allocation errors and
+-- application errors.
+handleComponentErrors :: IO a -> IO (Either ComponentError a)
+handleComponentErrors action = do
```

Comment:
This function seems to be kinda useless, as `runResourceAllocation` does it manually (for reasons).

Reply:
Right :man_facepalming: